doc/channel-plugin.md

```diff
@@ -0,0 +1,58 @@
+## Channel Plugin
+
+Channel plugins are processes that run independently of each other.
+
+Channel plugin uses stdin and stdout as json format to communicate. The request and response structure is inspired by
+JSON-RPC.
+
+### Request
+
+The request must be in json format and should contain following keys:
+
+- `method`: The request method to call
+- `params`: The params for request method
+- `id`: Unsigned int value. Require to assign the response to it's request as responses can be out of order
+
+Example:
+
+```
+{
+    "method": "Add",
+    "params": {
+        "num1": 5,
+        "num2": 3
+    },
+    "id" : 2020
+}
+```
+
+The example json files can be seen under `internal/channel/examples/`.
+
+### Response
+
+The response is in json format and contains following keys:
+
+- `result`: The result as json format, omitted if an error has occurred or the method does not return (e.g. setter calls)
+- `error`: The error message, omitted if no error occurred
+- `id`: The request id
+
+Example:
+
+```
+{
+    "result": 8,
+    "id" : 2020
+}
+```
+
+### Methods
+
+Currently, the channel plugin include following three methods:
+
+- `SetConfig`: Initialise the channel plugin with given config. The `params` key should contain the config. On success the response json only contains request id, ad
```

Comment:
"ad"? That sentence looks unfinished.

Should also say that the config is specific to the plugin and that the plugin defines what it expects there.